torch/testing/_internal/distributed/rpc/dist_autograd_test.py

```diff
@@ -2712,17 +2796,26 @@ def test_dist_autograd_sync_streams(self):
         remote_compute = rpc.remote(dst, TensorPipeCudaDistAutogradTest.MyRemoteCompute)
         local_compute = TensorPipeCudaDistAutogradTest.MyLocalCompute(remote_compute)
         for _ in range(10):
-            input = torch.rand([1000, 10000], device=self.rank, requires_grad=True)
+            if sparse:
+                input = build_sparse_tensor(requires_grad=True, device=self.rank)
+            else:
+                input = torch.rand([1000, 10000], device=self.rank, requires_grad=True)
             # Run local autograd
             result = input * 2.0
             r = random.random()
-            loss = result.sum() * r
+            if sparse:
+                loss = torch.sparse.sum(result) * r
+            else:
+                loss = result.sum() * r
             loss.backward()
 
             # Run distributed autograd
             with dist_autograd.context() as context_id:
                 result = local_compute(input)
-                loss = result.sum() * r
+                if sparse:
```

Comment:
This if-sparse-else has occur many times in this test file, shall we either extract that to a function or simplify it as 

```python
def my_sum(t):
  return torch.sparse.sum(result) if result.is_sparse else result.sum()

# or

torch.sparse.sum(result) if result.is_sparse else result.sum()
```

Reply:
I'll extract it to a function.